hypothesis-python/tests/cover/test_direct_strategies.py

```diff
@@ -154,8 +154,12 @@ def fn_ktest(*fnkwargs):
                      'blacklist_categories': ['Nd']}),
 )
 def test_validates_keyword_arguments(fn, kwargs):
+    from hypothesis.errors import HypothesisDeprecationWarning
     with pytest.raises(InvalidArgument):
-        fn(**kwargs).example()
+        # pytest.warns filters out the warning. It doesn't ensure that it
```

Comment:
This did seem weird... 🤔 And, I messed around with it, and I think there is a bit of an edge case here, when `pytest.warns` and `pytest.raises` are nested...

```python
import warnings
import pytest

def should_warn(): # but doesn't
    raise KeyError('...')

with pytest.raises(KeyError):
    with pytest.warns(UserWarning):
        warns()
```

This doesn't raise any error (python `3.7.0`, pytest `3.8.2`)

But when I tested:

```python
def should_warn():
    pass

with pytest.warns(UserWarning):
    warn()
```

This does raise an error.

What am I missing here?

> You shouldn't need to change this test at all though! If any entries use floats for a size argument, delete them here - you've already written the new tests with @checks_deprecated_behaviour instead.

You mean that I should change the test data? Wouldn't that break the compatibility? People would expect to get an `InvalidArgument`, yet we wouldn't be *testing* it (although it would still work). 

Anyway, this is resolved with your comment below 😄